### ISSUE TITLE
Menu button in example is missing an accessible name

### ISSUE THREAD

**Issue by mfairchild365** [angular/material2#15039]
#### What is the expected behavior?
The purpose of the menu button in the "menu with icons" example must be conveyed to users of screen reader software.

#### What is the current behavior?
The purpose of the menu button in the "menu with icons" example is not conveyed to users of screen reader software.

#### What are the steps to reproduce?
1. Using either of the following screen readers
   - NVDA 2018.4.1 + Firefox 64
   - VoiceOver for IOS (12.1.3) + Safari
2. Open the [menu example](https://material.angular.io/components/menu/examples)
3. Navigate to the menu button in the "menu with icons" example
   - In NVDA you can press the tab key until you reach the menu button
   - In VoiceOver for iOS you can swipe right until you reach the menu button
4. Observe that the menu button is announced as something like "menu button subMenu". In this case, "menu button" is the role of the element, not the name, and as such, the name of the menu button is not announced by the screen reader. Contrast this menu button to the menu button "basic menu" example, which is announced as "Menu, menu button subMenu". The name should be used to distinguish the menu buttom from other menu buttons on the page.

#### Which versions of Angular, Material, OS, TypeScript, browsers are affected?
- Angular Material Example 7.2.2 located at https://material.angular.io/components/
- NVDA 2018.4.1 + FireFox 64
- VoiceOver for IOS (12.1.3) + Safari (12.1.3)

#### Is there anything else we should know?
The [documentation for the menu component](https://material.angular.io/components/menu/overview) states

> Menu triggers or menu items without text or labels should be given a meaningful label via `aria-label` or `aria-labelledby`

The example should be updated to follow that advice.
